File: copy/backend/src/main/java/com/capstone/backend/service/UserService.java
package com.capstone.backend.service;

import com.capstone.backend.model.Users;
import com.capstone.backend.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class UserService {
    @Autowired
    UserRepository userRepository;

    public List<Users> getAll() {
        return userRepository.findAll();
    }

    public Users getUserById(int userId) {
        return userRepository.getUserById(userId);
    }

    public Users getByUserName(String userName) {
        return userRepository.getByUserName(userName);
    }

    public String addUser(Users use) {
        String response;

        if (userRepository.addUser(use))
            response = "Successfully added";
        else
            response = "Something went wrong. Not added, please try again";
        return response;
    }

    public String updateUser(int userId, Users use) {
        String response;

        if (userRepository.updateUser(userId, use))
            response = "Successfully updated";
        else
            response = "Something went wrong. Not updated, please try again";
        return response;
    }

    public String deleteUser(int userId) {
        String response;

        if (userRepository.deleteUser(userId))
            response = "Successfully deleted";
        else
            response = "Something went wrong. Not deleted, please try again";
        return response;
    }

    public ResponseEntity<?> login(Users user) {
        Users userdata = userRepository.getByUserName (user.getUserName()); //VERIFY IS USER EXISTS
        if(userdata != null)
        {
            if(user.getUserPassword ().equals(userdata.getUserPassword ()))
            {
                return new ResponseEntity<Users> (userdata, HttpStatus.OK);
            }
            else
            {
                return new ResponseEntity<> ("Wrong Password", HttpStatus.BAD_REQUEST);
            }
        }
        else
        {
            return new ResponseEntity<>(  "User does not exist!", HttpStatus.BAD_REQUEST);
        }
    }

    public boolean register(Users user) {
        return userRepository.register(user);
    }

}

File: copy/frontend/src/services/UserService.js
import axios from 'axios'

const USER_BASE_REST_API_URL='http://localhost:8080/api/v1/users';

class UserService{
    
    getAllUsers(){
        return axios.get(USER_BASE_REST_API_URL + '/getAll') //get method 
    }
    createUser(user){
        return axios.post(USER_BASE_REST_API_URL + '/addUser', user) //post method
    }
    getUserbyId(userId){
        return axios.get(USER_BASE_REST_API_URL + '/getUserById/' + userId) //get method
    }
    updateUser(userId, user){
        return axios.put(USER_BASE_REST_API_URL + '/updateUser/' + userId, user) //put method
    }
    deleteUser(userId){
        return axios.delete(USER_BASE_REST_API_URL+ '/deleteUser/' + userId) //delete method
    }
    loginUser(user){
        return axios.post('http://localhost:8080/api/v1/login/', user) //delete method
    }
}

export default new UserService();